### DISCUSSION TITLE
Pass obj argument to callback funcs

### DISCUSSION THREAD

**Discussion by velazquezr35** [versa-syahptr/winotify#17] [Q&A]
Hello there,

Is it possible to pass custom arguments to callback functions? All examples are simple, with some print or global definitions, but not with object handling (for example alter X attribute when calling the callback).

Thanks :)

**Comment by velazquezr35** [comment]
This seems to work, can there be any drawbacks?

So, it is possible to edit the internal object attribute from the callback:

```
def upper(arg1):
    def mid(func):
        def wrap():
            return func(arg1)
        return wrap
    return mid

def mf(obj):
    print(f'Mark as C for: {obj.name}')
    obj.status = 'C'

class sample_obj():
    def __init__(self, name):
        self.name = name

def main(p):
    @notifier.register_callback
    def x():
        return upper(p)(mf)() 

    toast = notifier.create_notification("a notification", 'a notification test with callback')#,
                                         # launch=notifier.callback_to_url(do_somethin))
    # toast.add_actions("Open Github", "https://github.com/versa-syahptr/winotify")
    # toast.add_actions("Quit app", quit_)
    # toast.add_actions("spam", spam)
    toast.add_actions('spam', x)
    toast.show()


@notifier.register_callback
def clear(): notifier.clear()


if __name__ == '__main__':
    p = sample_obj('pedro')
    notifier.start()
    main(p)
```

**Comment by versa-syahptr** [comment]
Hey, it's been a while since I worked on this. 
It's actually possible to pass arguments to the callback, but only python literal types can be used. Before version `1.1.0` was released, I've tried to make it but only strings works, so I decided not to release it. I'll try to make it for the next version

Another workaround maybe you can use lambda function or `functools.partial`

```python
cb = functools.partial(mf, p)
notifier.register_callback(cb)
...
toast.add_actions('spam', cb)
```
